src/PluginClient.php

```diff
@@ -79,7 +78,7 @@ public function sendRequest(RequestInterface $request)
         $pluginChain = $this->createPluginChain($this->plugins, function (RequestInterface $request) {
             try {
                 return new FulfilledPromise($this->client->sendRequest($request));
-            } catch (HttplugException $exception) {
+            } catch (\Exception $exception) {
```

Comment:
This also indicates a BC break: https://github.com/php-http/httplug/blob/master/src/HttpAsyncClient.php#L24

Meaning if the exception is not an HTTP one, it should be thrown immediately.

So to sum up: the Promise actually allows to receive a `\Exception` instance in the rejection callback, however Promises in the HTTP flow should only receive HTTP related exceptions, everything else should immediately be thrown.